File: ios/Classes/handler/CommandHandler.swift
import Flutter
import Foundation

enum MethodName: String {
    case trimVideo = "trimVideo"
    case mergeVideos = "mergeVideos"
    case extractAudio = "extractAudio"
    case adjustVideoSpeed = "adjustVideoSpeed"
    case removeAudio = "removeAudio"
    case cropVideo = "cropVideo"
    case rotateVideo = "rotateVideo"
    case generateThumbnail = "generateThumbnail"
    case compressVideo = "compressVideo"
    case cancelOperation = "cancelOperation"
    case getVideoMetadata = "getVideoMetadata"
    case flipVideo = "flipVideo"
    case maxFps = "maxFps"
}

class CommandHandler {
    private let handlers: [MethodName: Command]
    
    init() {
        handlers = [
            .trimVideo: TrimVideoCommand(),
            .mergeVideos: MergeVideosCommand(),
            .extractAudio: ExtractAudioCommand(),
            .adjustVideoSpeed: AdjustVideoSpeedCommand(),
            .removeAudio: RemoveAudioCommand(),
            .cropVideo: CropVideoCommand(),
            .rotateVideo: RotateVideoCommand(),
            .generateThumbnail: GenerateThumbnailCommand(),
            .compressVideo: CompressVideoCommand(),
            .cancelOperation: CancelOperationCommand(),
            .getVideoMetadata: GetVideoMetadataCommand(),
            .flipVideo: FlipVideoCommand(),
            .maxFps: MaxFpsCommand()
        ]
    }
    
    func handleCommand(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
            guard let methodName = MethodName(rawValue: call.method) else {
                result(FlutterMethodNotImplemented)
                return
            }
            
            if let handler = handlers[methodName] {
                handler.execute(call: call, result: result)
            } else {
                result(FlutterMethodNotImplemented)
            }
        }
    }

File: android/src/main/kotlin/com/example/easy_video_editor/handler/CommandHandler.kt
package com.example.easy_video_editor.handler

import android.content.Context
import com.example.easy_video_editor.command.Command
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

enum class MethodName(val methodName: String) {
    TRIM_VIDEO("trimVideo"),
    MERGE_VIDEOS("mergeVideos"),
    EXTRACT_AUDIO("extractAudio"),
    ADJUST_VIDEO_SPEED("adjustVideoSpeed"),
    REMOVE_AUDIO("removeAudio"),
    CROP_VIDEO("cropVideo"),
    ROTATE_VIDEO("rotateVideo"),
    GENERATE_THUMBNAIL("generateThumbnail"),
    COMPRESS_VIDEO("compressVideo");

    companion object {
        fun fromString(method: String): MethodName? {
            return values().find { it.methodName == method }
        }
    }
}

class CommandHandler (private val context: Context) {
    private val handlers: Map<MethodName, Command> = mapOf(
        MethodName.TRIM_VIDEO to TrimVideoCommand(context),
        MethodName.MERGE_VIDEOS to MergeVideosCommand(context),
        MethodName.EXTRACT_AUDIO to ExtractAudioCommand(context),
        MethodName.ADJUST_VIDEO_SPEED to AdjustVideoSpeedCommand(context),
        MethodName.REMOVE_AUDIO to RemoveAudioCommand(context),
        MethodName.CROP_VIDEO to CropVideoCommand(context),
        MethodName.ROTATE_VIDEO to RotateVideoCommand(context),
        MethodName.GENERATE_THUMBNAIL to GenerateThumbnailCommand(context),
        MethodName.COMPRESS_VIDEO to CompressVideoCommand(context)
    )

    fun handleCommand(call: MethodCall, result: MethodChannel.Result) {
        val methodName = MethodName.fromString(call.method)
        if (methodName == null) {
            result.notImplemented()
            return
        }
        handlers[methodName]?.execute(call, result) ?: result.notImplemented()
    }
} 